### DISCUSSION TITLE
Github: Repo not found

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [statiqdev/Discussions#81] [General]
In my Development.yml workflow definition, I have the following action defined:

```yml
# Build and push to GitHub Pages
- name: build site
  run: dotnet run -- deploy --development -l Debug
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

In Program.cs, I have this to tie it together:

```cs
if (args.Contains("--development"))
{
    bootstrapper.DeployToGitHubPages(
        "LuzFaltex",
        "luzfaltex.com",
        Config.FromSetting<string>("GITHUB_TOKEN")
    );
}
```

This appears to work, except I get a "not found" error from the GitHub integration:

```
[INFO] Deployment/Output » ExecuteIf » LogMessage » Deploying to GitHub repository LuzFaltex/luzfaltex.com
[DBUG] Deployment/Output » ExecuteIf » LogMessage » Finished module execution (0 output document(s), 2 ms)
[DBUG] Deployment/Output » ExecuteIf » ExecuteIf » Starting module execution... (0 input document(s))
[DBUG] Deployment/Output » ExecuteIf » ExecuteIf » DeployGitHubPages » Starting module execution... (0 input document(s))
[ERRO] Deployment/Output » ExecuteIf » ExecuteIf » DeployGitHubPages » Not Found
[DBUG] Exception while executing pipeline Deployment/Output: Octokit.NotFoundException: Not Found
```

I've experimented with several things, but in the end I decided to just pull open Postman and ask GitHub if it exists:

```
GET: https://api.github.com/repos/LuzFaltex/luzfaltex.com?type=public&sort=full_name```

```json
{
    "id": 328477771,
    "node_id": "MDEwOlJlcG9zaXRvcnkzMjg0Nzc3NzE=",
    "name": "luzfaltex.com",
    "full_name": "LuzFaltex/luzfaltex.com",
    "private": false,
// snip
}
```

GitHub seems to think it exists, but OctoCat does not. Is there a reason behind this disparity?

**Comment by daveaglick** [comment]
For anyone else who ends up here, GitHub Pages deployment problems are usually one of two things:
- The branch name is wrong - the default branch for GitHub Pages deployment is "gh-pages", though it can be adjusted following the instructions and settings here: https://statiq.dev/web/deployment/github-pages
- The branch is uninitialized - a deployment won't be able to find the branch if it exists, but is uninitialized (this is discussed in more detail at https://docs.github.com/en/pages/getting-started-with-github-pages/creating-a-github-pages-site).

However, that doesn't seem to be your specific problem. In looking at the site I see it does have a "gh-pages" branch and the branch is initialized. The next thing that comes to mind as a potential problem is the access token, though that looks okay too.

In looking at the configuration of your GitHub Actions and the error messages, I'm seeing the not found error show up for the recent Actions runs of the `Development` action. In the `.github/workflows` folder I see a YAML file `Development.yml` that I assume is the YAML for the Action that's displaying the error? In that YAML file I don't actually see the `GITHUB_TOKEN` secret being injected. Is it possible there's a mixup between the YAML definition and the Action being run?

**Reply by TheBrambleShark** [reply]
When I push to the dev branch, it seems it is using the Development.yml file from that branch to run the action.

Actually my error was something I had missed several times and it seems you had missed too, but repeating your steps allowed me to spot it.

My pages branch is `gh_pages` instead of `gh-pages`. Changing my Program.cs to this seems to have resolved the issue:

```cs
bootstrapper.DeployToGitHubPagesBranch(
    "LuzFaltex",
    "luzfaltex.com",
    Config.FromSetting<string>("GITHUB_TOKEN"),
    "gh_pages"
);
```

**Reply by daveaglick** [reply]
Oh good, another problem down! I never would have caught that difference either - good find. To help these situations going forward, I can add some extra logging to the output - that might help verify site, owner, branch, etc. information in the future.
